src/phoenix/trace/openai/instrumentor.py

```diff
@@ -616,8 +626,12 @@ def _accumulate_messages(
     if not chunks:
         return []
     content_token_lists: DefaultDict[int, List[str]] = defaultdict(list)
-    function_argument_token_lists: DefaultDict[int, List[str]] = defaultdict(list)
-    function_names: Dict[int, str] = {}
+    function_calls: DefaultDict[int, StreamingFunctionCallData] = defaultdict(
+        lambda: StreamingFunctionCallData(argument_tokens=[])
+    )
+    tool_calls: DefaultDict[int, DefaultDict[int, StreamingFunctionCallData]] = defaultdict(
+        lambda: defaultdict(lambda: StreamingFunctionCallData(argument_tokens=[]))
```

Comment:
nit. i think using dataclass can save you a few keystrokes
```suggestion
        lambda: defaultdict(StreamingFunctionCallData))
```

Reply:
Addressed in d9112733